src/modules/Game.class.js

```diff
@@ -1,68 +1,301 @@
+            newRow[i] *= 2;
+            this.score += newRow[i];
+            newRow.splice(i + 1, 1);
+            merged[i] = true;
+          }
+        }
+
+        while (newRow.length < size) {
+          newRow.push(0);
+        }
+
+        for (let col = 0; col < size; col++) {
+          if (state[row][col] !== newRow[col]) {
+            state[row][col] = newRow[col];
+            moved = true;
+          }
+        }
+      }
+
+      this.handleMove(moved);
+    }
+  }
+
+  moveRight() {
+    if (this.getStatus() === Game.STATUS.PLAYING) {
+      const size = this.state.length;
+      const state = this.getState();
+      let moved = false;
+
+      for (let row = 0; row < size; row++) {
+        const newRow = [];
+
+        for (let col = 0; col < size; col++) {
+          if (state[row][col] !== 0) {
+            newRow.push(state[row][col]);
+          }
+        }
+
+        const merged = Array(newRow.length).fill(false);
+
+        for (let i = newRow.length - 1; i > 0; i--) {
+          if (newRow[i] === newRow[i - 1] && !merged[i] && !merged[i - 1]) {
+            newRow[i] *= 2;
+            this.score += newRow[i];
+            newRow.splice(i - 1, 1);
+            merged.splice(i - 1, 1);
+            merged[i - 1] = true;
+          }
+        }
+
+        while (newRow.length < size) {
+          newRow.unshift(0);
+        }
+
+        for (let col = 0; col < size; col++) {
+          if (state[row][col] !== newRow[col]) {
+            state[row][col] = newRow[col];
+            moved = true;
+          }
+        }
+      }
+
+      this.handleMove(moved);
+    }
   }
 
-  moveLeft() {}
-  moveRight() {}
-  moveUp() {}
-  moveDown() {}
-
-  /**
-   * @returns {number}
-   */
-  getScore() {}
-
-  /**
-   * @returns {number[][]}
-   */
-  getState() {}
-
-  /**
-   * Returns the current game status.
-   *
-   * @returns {string} One of: 'idle', 'playing', 'win', 'lose'
-   *
-   * `idle` - the game has not started yet (the initial state);
-   * `playing` - the game is in progress;
-   * `win` - the game is won;
-   * `lose` - the game is lost
-   */
-  getStatus() {}
-
-  /**
-   * Starts the game.
-   */
-  start() {}
-
-  /**
-   * Resets the game.
-   */
-  restart() {}
-
-  // Add your own methods here
+  moveUp() {
+    if (this.getStatus() === Game.STATUS.PLAYING) {
+      const size = this.state.length;
+      const state = this.getState();
+      let moved = false;
+
+      for (let col = 0; col < size; col++) {
+        const newColumn = [];
+
+        for (let row = 0; row <= size - 1; row++) {
+          if (state[row][col] !== 0) {
+            newColumn.push(state[row][col]);
+          }
+        }
+
+        for (let i = 0; i < newColumn.length; i++) {
+          if (newColumn[i] === newColumn[i + 1]) {
+            newColumn[i] *= 2;
+            this.score += newColumn[i];
+            newColumn.splice(i + 1, 1);
+          }
```

Comment:
You have a potential bug here in 'moveUp': the merged array is not being used, which could lead to incorrect merging behavior. Tiles might merge more than once per move, which is not allowed in the game rules.